tools/skbstat_example.txt

```diff
@@ -0,0 +1,82 @@
+Demonstrations of skbstat, the Linux BPF/bcc version.
+
+
+skbstat prints detailed information about the packet (skb) processed by the kernel,
+including ingress/egress/drop. we can pass filter parameters to it, such as "-p tcp -P 80":
+
+# ./skbstat.py -p tcp -P 80 --input --output
+TIME     NUM    FUNC/STACK
+20:04:29        [INGRESS]
+         1      tcp_v4_rcv
+         1      ip_local_deliver_finish
+         1      ip_local_deliver
+         1      ip_route_input_noref
+         1      ip_rcv_finish
+         1      ip_rcv
+         1      netif_receive_skb_internal
+
+TIME     NUM    FUNC/STACK
+20:04:30        [EGRESS]
+         1      ip_output
+         1      ip_finish_output
+         1      ip_finish_output2
+         1      __dev_queue_xmit
+         1      dev_hard_start_xmit
+
+TIME     NUM    FUNC/STACK
+20:04:30        [DROP]
+         1
+                kfree_skb+0x1
+                tcp_v4_rcv+0x164
+                ip_protocol_deliver_rcu+0x36
+                ip_local_deliver_finish+0x44
+                ip_local_deliver+0x6b
+                ip_rcv+0x52
+                __netif_receive_skb_one_core+0x50
+                netif_receive_skb_internal+0x2f
+                napi_gro_receive+0x107
+
+[...]
+
+The last column shows the number of function calls or skb dropped stacks.
+
+This tool is useful for debugging why a particular packet (skb) was dropped.
```

Comment:
Is it possible tcp_drop itself (tcpdrop.py) would achieve the same purpose? Any other use cases?

Reply:
the drop of none TCP packet is also hard to debug, and `skbstat` can filter some unrelated/uncared packet for us.